src/opnsense/mvc/app/controllers/OPNsense/Base/ControllerBase.php

```diff
@@ -204,5 +204,10 @@ public function beforeExecuteRoute($dispatcher)
 
         // append ACL object to view
         $this->view->acl = new \OPNsense\Core\ACL();
+        $this->response->setHeader('Content-Security-Policy', "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline' 'unsafe-eval';");
+        $this->response->setHeader('X-Frame-Options', "SAMEORIGIN");
```

Comment:
Allow framing only in the context of the firewall itself (no framing by another website) - avoid clickjacking

Reply:
@fabianfrz this one is already in there https://github.com/opnsense/core/blob/18.1.2/src/opnsense/mvc/app/controllers/OPNsense/Base/IndexController.php#L77 , we could move it to Base and make sure the classes deriving from index move to base. What do you think?

Reply:
So it should be removed from the index controller since it inherits controllerbase and security should be enforced on all controllers - even those which are not IndexControllers.